lib/grpcxx/uv/server.cpp

```diff
@@ -72,15 +68,10 @@ void server::prepare(std::string_view ip, int port) {
 	start_listening();
 }
 
-void server::prepare(int fd) {
-	struct sockaddr_in addr;
-	socklen_t          addr_len = sizeof(addr);
-	if (auto r = uv_tcp_open(&_handle, fd);
-		r != 0 || getsockname(fd, (struct sockaddr *)&addr, &addr_len) != 0) {
```

Comment:
`getsockname()` will require different headers on *nix and windows. Potentially we could use [uv_tcp_getsockname()](https://docs.libuv.org/en/v1.x/tcp.html#c.uv_tcp_getsockname) but it'll still require a platform specific header for `sockaddr_storage`, unless we rely on `<uv.h>` to include the headers needed.

@tchernobog I feel it's acceptable to not have the extra validation here for now since this is a protected member. Thoughts?

Reply:
The problem is that it is protected, but it is called by `run()` which is not. I would use `uv_tcp_getsockname()` and keep the check myself